Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/internal/Snapping/snap.h

```diff
@@ -0,0 +1,406 @@
+{
+  using boost::get_param;
+  using boost::choose_param;
+
+  typedef typename boost::graph_traits<PolygonMesh>::vertex_descriptor      vertex_descriptor;
+  typedef typename boost::graph_traits<PolygonMesh>::halfedge_descriptor    halfedge_descriptor;
+  typedef typename boost::graph_traits<PolygonMesh>::edge_descriptor        edge_descriptor;
+
+  typedef CGAL::Box_intersection_d::Box_with_info_d<double, 3, vertex_descriptor>     Box;
+
+  typedef typename GetVertexPointMap<PolygonMesh, SourceNamedParameters>::type        SVPM;
+  typedef typename GetVertexPointMap<PolygonMesh, TargetNamedParameters>::const_type  TVPM;
+  typedef typename boost::property_traits<TVPM>::value_type                           Point;
+
+  SVPM svpm = choose_param(get_param(snp, internal_np::vertex_point),
+                           get_property_map(vertex_point, smesh));
+
+  TVPM tvpm = choose_param(get_param(tnp, internal_np::vertex_point),
+                           get_const_property_map(vertex_point, tmesh));
+
+  typedef typename GetGeomTraits<PolygonMesh, SourceNamedParameters>::type            GT;
+  GT gt = choose_param(get_param(snp, internal_np::geom_traits), GT());
+
+  CGAL_static_assertion((boost::is_same<Point, typename GT::Point_3>::value));
+
+  typedef typename GT::FT                                                             FT;
+  typedef CGAL::internal::Dynamic_property_map<vertex_descriptor, FT>       Default_tolerance_map;
+
+  typedef typename boost::lookup_named_param_def <
+    internal_np::tolerance_map_t,
+    SourceNamedParameters,
+    Default_tolerance_map // default
+  > ::type                                                                  Tolerance_map;
+
+  Tolerance_map tol_vm = choose_param(get_param(snp, internal_np::tolerance_map),
+                                      Default_tolerance_map());
+
+  // Extract border vertices
+  std::vector<halfedge_descriptor> border_vertices;
+  BOOST_FOREACH(edge_descriptor ed, edges(smesh))
+  {
+    if(is_border(ed, smesh))
+    {
+      halfedge_descriptor hd = halfedge(ed, smesh);
+      if(!is_border(hd, smesh))
+        hd = opposite(hd, smesh);
+      border_vertices.push_back(hd);
+    }
+  }
+
+  // If no tolerance is provided, compute a unique bound at each vertex
+  if(boost::is_same<Tolerance_map, Default_tolerance_map>::value)
+    compute_tolerance_at_vertices(tol_vm, border_vertices, svpm, gt, smesh);
+
+  // Try to snap vertices
+  std::vector<Box> boxes;
+  BOOST_FOREACH(halfedge_descriptor hd, border_vertices)
+  {
+    vertex_descriptor vd = target(hd, smesh);
+    const FT eps = get(tol_vm, vd);
+
+    const typename GT::Vector_3 eps_v = gt.construct_vector_3_object()(eps, eps, eps);
+    const typename GT::Vector_3 opp_eps_v = gt.construct_vector_3_object()(-eps, -eps, -eps);
+    const Point& pm = gt.construct_translated_point_3_object()(get(svpm, vd), opp_eps_v);
+    const Point& pM = gt.construct_translated_point_3_object()(get(svpm, vd), eps_v);
+
```

Comment:
Why not directly building the bbox out of `p` and `eps`? This would simplify the code and the doc.

Reply:
This comment is still valid.